src/gosling-genomic-axis/axis-track.ts

```diff
@@ -20,12 +20,19 @@ type TickLabelInfo = {
     rope: PIXI.SimpleRope;
```

Comment:
I think I had to make this optional:

```suggestion
    rope?: PIXI.SimpleRope;
```

Reply:
Related to my comment below.

Reply:
It seems like this can also be `null` though, since `this.addCurvedText(...)` can return `null`:


```typescript
        addCurvedText(textObj: any, cx: number) {
// ...
             if (ropePoints.length === 0) {
                return null;
            }

```

Reply:
So the type would need to be:

```typescript
    rope?: PIXI.SimpleRope | null;
```

 to allow this code to type-check completely. Allowing a field to be `null` | `undefined` is a pattern I really try to avoid since it leads to complicated interfaces like this, and semantically `null` and `undefined` usually mean the same thing. My personal preference is to avoid `null` in JavaScript, since we have also have `undefined` (nice post here on topic https://github.com/sindresorhus/meta/discussions/7), but I'm fine with choosing one or the other.

Reply:
I agree. I am also trying to avoid using `null`. We can return `undefined` instead of adding `| null` to the type:

```ts
if (ropePoints.length === 0) {
                return undefined;
            }
```

Reply:
Ok great!